_misc/sanity_utils.py

```diff
@@ -0,0 +1,151 @@
+import fnmatch
+import os
+import posixpath
+import re
+from ast import iter_fields, AST, Attribute, Name
+import six
+
+IGNORED_DIRS = [
+    '*.egg-info',
+    '.bzr',
+    '.cache',
+    '.git',
+    '.hg',
+    '.idea',
+    '.svn',
+    '.tox',
+    '__pycache__',
+    '_build',
+    'bower_components',
+    'CVS',
+    'htmlcov',
+    'node_modules',
+    'venv*',
+]
+
+IGNORED_PATH_REGEXPS = [
+    '^.*shoop/admin/static/shoop_admin.*',  # autogenerated
+    '^.*shoop/xtheme/static/*',  # autogenerated
+    '^build/',  # built files
+]
+
+IGNORED_PATTERNS = [
+    '*-bundle.js',
+    '*.bat',
+    '*.bz2',
+    '*.dat',
+    '*.doctree',
+    '*.eot',
+    '*.gif',
+    '*.gz',
+    '*.ico',
+    '*.inv',
+    '*.jpg',
+    '*.map',  # source map
+    '*.min.js',
+    '*.mo',
+    '*.otf',
+    '*.pickle',
+    '*.png',
+    '*.pyc',
+    '*.sqlite3',
+    '*.svg',
+    '*.ttf',
+    '*.woff',
+    '*.woff2',
+    '*.zip',
+    '.coverage',
+    '_version.py',
+    'Makefile',
+    'vendor.js',
+]
+
+
+def find_files(
+        roots, generated_resources=None,
+        ignored_patterns=IGNORED_PATTERNS,
+        ignored_dirs=IGNORED_DIRS,
+        ignored_path_regexps=IGNORED_PATH_REGEXPS,
+        allowed_extensions=None
+):
+    """
+    Find files in `roots` with ignores, `generated_resources` handling etc.
+
+    :param roots: Root directory or directories
+    :type roots: str
+    :param generated_resources: Output set of generated resources (mutated during find_files)
+    :type generated_resources: set
+    :param ignored_patterns: fnmatch file patterns to ignore
+    :type ignored_patterns: Iterable[str]
+    :param ignored_dirs: fnmatch directory patterns to ignore
+    :type ignored_dirs: Iterable[str]
+    :param ignored_path_regexps: Path regexps to ignore
+    :type ignored_path_regexps: Iterable[str]
+    :param allowed_extensions: Extensions (really filename suffixes) to ignore (optional)
+    :type allowed_extensions: Iterable[str]|None
+    :return: Iterable of file paths
+    :rtype: Iterable[str]
+    """
+    if generated_resources is None:
+        generated_resources = set()
+    if isinstance(roots, six.string_types):
+        roots = [roots]
+    if isinstance(allowed_extensions, six.string_types):
+        allowed_extensions = set([allowed_extensions])
```

Comment:
Don't use six, since that's not available at our Jenkins host while doing sanity checks.